File: rileylink/src/main/java/info/nightscout/androidaps/plugins/pump/common/hw/rileylink/defs/RileyLinkPumpDevice.java
package info.nightscout.androidaps.plugins.pump.common.hw.rileylink.defs;

import info.nightscout.androidaps.plugins.pump.common.hw.connector.defs.CommunicatorPumpDevice;
import info.nightscout.androidaps.plugins.pump.common.hw.rileylink.service.RileyLinkService;

public interface RileyLinkPumpDevice extends CommunicatorPumpDevice {

    void setBusy(boolean busy);

    void triggerPumpConfigurationChangedEvent();

    RileyLinkService getRileyLinkService();

    RileyLinkPumpInfo getPumpInfo();

    long getLastConnectionTimeMillis();

    void setLastCommunicationToNow();
}

File: pump/rileylink/src/main/java/info/nightscout/androidaps/plugins/pump/common/hw/rileylink/defs/RileyLinkPumpDevice.kt
package info.nightscout.androidaps.plugins.pump.common.hw.rileylink.defs

import info.nightscout.androidaps.plugins.pump.common.hw.rileylink.service.RileyLinkService

interface RileyLinkPumpDevice {

    fun setBusy(busy: Boolean)
    fun triggerPumpConfigurationChangedEvent()
    val rileyLinkService: RileyLinkService?
    val pumpInfo: RileyLinkPumpInfo
    val lastConnectionTimeMillis: Long
    fun setLastCommunicationToNow()
}